src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractAccessControlNameCheck.java

```diff
@@ -95,6 +96,22 @@ protected boolean shouldCheckInScope(DetailAST modifiers) {
                 || applyToPrivate && isPrivate;
     }
 
+    /**
+     * Checks if given modifiers has public access.
+     * There are 2 cases - it is either has explicit modifier, or it is
+     * in annotation or interface.
+     *
+     * @param modifiers - modifiers to check
+     * @return true if public
+     */
+    private static boolean isPublic(DetailAST modifiers) {
+        return ScopeUtil.isInInterfaceBlock(modifiers)
+                // interface methods can be private
```

Comment:
methods? this check is only about fields.

Reply:
This method is common for several checks, including MethodNameCheck. I added test case, report and PR title shows that this one is also fixed.